src/ray/raylet/task_dependency_manager.h

```diff
@@ -120,14 +128,18 @@ class TaskDependencyManager {
   std::string DebugString() const;
 
  private:
-  using ObjectDependencyMap = std::unordered_map<ray::ObjectID, std::vector<ray::TaskID>>;
+  using ObjectDependencyMap =
+      std::unordered_map<ray::ObjectID, std::unordered_set<ray::TaskID>>;
 
   /// A struct to represent the object dependencies of a task.
   struct TaskDependencies {
     /// The objects that the task is dependent on. These must be local before
     /// the task is ready to execute.
-    std::unordered_set<ObjectID> object_dependencies;
-    /// The number of object arguments that are not available locally. This
+    std::unordered_set<ObjectID> get_dependencies;
+    /// The objects that the task is fetching. These are fetched while the
+    /// the task is executing.
+    std::unordered_set<ObjectID> wait_dependencies;
```

Comment:
For `get_dependencies` and `wait_dependencies`, the semantics of when things are added/removed from those sets differs, right? Can we document how they differ?